### ISSUE TITLE
Can't use terrains with "call on use" ability

### ISSUE THREAD

**Issue by sylae** [calref/cboe#8]
_Original author: celtic.m...@gmail.com (May 23, 2009 02:47:11)_

<b>What steps will reproduce the problem?</b>
1.Make a terrain that calls a special node when used.
2.Play the game and attempt to use the terrain.

<b>What is the expected output? What do you see instead?</b>
The node should be called. Instead, you get the &quot;can't use&quot; error.

_Original issue: http://code.google.com/p/openexile/issues/detail?id=13_

**Comment by CelticMinstrel** [comment]
I'm not sure if this is still an issue; need to do some testing.

**Comment by NQNStudios** [comment]
I'm testing this and not getting the error described, but I *am* not seeing a Use button for the outdoors, which is a different problem that technically fits with the title of this issue.

**Comment by NQNStudios** [comment]
https://github.com/NQNStudios/cboe/blob/69d07f87ef0624c156c9569d4a8c4c1427b9962d/src/game/boe.actions.cpp#L2463-L2466

The keyboard shortcut also won't work outdoors. I'm guessing in the legacy game there was no Call on use in the outdoors, and it was added as a feature but Use outdoors was not?

**Comment by NQNStudios** [comment]
I suppose it could be affecting legacy scenarios, so I should take the extra time to make one and test those too.

**Comment by NQNStudios** [comment]
Oh, in legacy scenarios there is no option for "local" special nodes for this. So my theory appears correct -- someone was adding local node support and added outdoor support on the backend, but not frontend.

**Comment by NQNStudios** [comment]
Aha! With a legacy scenario it prints "Nothing to use". The bug is real

**Comment by NQNStudios** [comment]
For reasons that are not our problem, the legacy scenario editor wouldn't let me put "Call special when used" on certain terrains, but I found one that it did, exported the scenario, and tested again. You *can* use terrains and get the special node from both new and old format scenarios. I will PR my replay test and the scenedit changes I made along the way, and I'm closing this issue as already complete.
